src/app/clients/add/[id]/autofill.ts

```diff
@@ -0,0 +1,62 @@
+import { Schema } from "@/database_utils";
+import { ClientFields } from "@/app/clients/add/ClientForm";
+import { Person } from "@/components/Form/formFunctions";
+import { booleanGroup } from "@/components/DataInput/inputHandlerFactories";
+import { processExtraInformation } from "@/pdf/ShoppingList/dataPreparation";
+
+const getNumberAdultsByGender = (family: Schema["families"][], gender: string): number => {
+    return family.filter((member) => member.gender === gender).length;
+};
+
+const revertBooleanGroup = (data: string[]): booleanGroup => {
+    const reverted: booleanGroup = {};
+    data.forEach((value) => (reverted[value] = true));
+    return reverted;
+};
+
+const autofill = (
+    clientData: Schema["clients"],
+    familyData: Schema["families"][]
+): ClientFields => {
+    const children = familyData
+        .filter((member) => member.age !== null)
+        .map((child): Person => {
+            return {
+                gender: child.gender,
+                age: child.age,
+                quantity: 1,
+                primaryKey: child.primary_key,
+            };
+        });
+
+    const { nappySize, extraInformation } = processExtraInformation(clientData.extra_information);
+
+    return {
+        fullName: clientData.full_name,
+        phoneNumber: clientData.phone_number,
+        addressLine1: clientData.address_1,
+        addressLine2: clientData.address_2,
+        addressTown: clientData.address_town,
+        addressCounty: clientData.address_county,
+        addressPostcode: clientData.address_postcode,
+        adults: [
+            { gender: "other", quantity: getNumberAdultsByGender(familyData, "other") },
+            { gender: "male", quantity: getNumberAdultsByGender(familyData, "male") },
+            { gender: "female", quantity: getNumberAdultsByGender(familyData, "female") },
+        ],
+        numberChildren: children.length,
+        children: children,
+        dietaryRequirements: revertBooleanGroup(clientData.dietary_requirements),
```

Comment:
Just thinking if there is a more intuitive way to name the function
Was a bit lost of what is `revertBooleanGroup` until talking to you f2f